### ISSUE TITLE
Able to set default authentication method for login

### ISSUE THREAD

**Issue by ogarcia** [gogs/gogs#5274]
With the change introduced in #3142 now in the login screen appears a new option to select the source of authentication but there is not any method to select default source or hide this option.

I think that here can be two features.
1. The ability to select the default authentication method.
2. The ability to hide this option.

IMHO. I think that the behaviour may be that you can select a default method for authentication and, if cannot authenticate, the app may fallback always to local auth. With this method if you have only one backend for auth you can hide the option in login window that is more easy for the users.

**Comment by kidmock** [comment]
I would really like to see this feature. My users are complaining they can't log in, because they don't understand the "Authentication Source" dropdown. I would love to see this hidden

**Comment by louzadod** [comment]
I like the possibility to login even if my ldap service is down (I have a local admin user). So I don't want to hide the "Authentication Source" dropdown. 

However, define the default authentication source whould be great for the user experience.
+1

**Comment by chrisbrookes** [comment]
Perhaps we have a default but also hide the dropdown with a way to make it appear (small javascript link "change auth source") if needed? I'm assuming that once a default is set, 99% of the time the user would never need to see the dropdown.

**Comment by cleary** [comment]
There's a pull request that looks like it will do this (or something very close) here:
https://github.com/gogs/gogs/pull/5338

It could do with some upvotes :)

**Comment by DMW007** [comment]
For me this breaks the LDAP login completely in 0.11.53 so that I downgraded to 0.11.43 to make it work. It doesn't make any difference if local or LDAP is selected, the login on my ApacheDS DC is not possible any more.  Credentials are correct since after the downgrade I can do a LDAP login with the same login data.

**Comment by crd477** [comment]
I recently worked around this in my own environment with the following:
```
$ diff -br templates/user/auth/login.tmpl custom/templates/user/auth/login.tmpl
24c24
<                                                               <input type="hidden" id="login_source" name="login_source" value="{{.login_source}}" required>
---
>                                                               <input type="hidden" id="login_source" name="login_source" value="1" required>
```
where "1" is the database id of the `login_source` I want to be the default.

The advantage of doing this is that it's a trivial change leveraging the elegant Go/Gogs templating system and does not prevent users from using the dropdown if they still need to.

This disadvantage of doing this is that it copies a whole file for one small change and it's probably not the cleanest in terms of being upgrade safe.

**Comment by Unknwon** [comment]
This is claimed to be fixed by merging #5338, please help test on `develop` branch.
